Src/Support/IntegrationTests/HttpClientFromMessageHandlerFactoryTests.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if NETCOREAPP2_1
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using System.Net.Http;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Net;
+using Google.Apis.Http;
+using Google.Apis.Translate.v2;
+using Google.Apis.Auth.OAuth2;
+using IntegrationTests.Utils;
+using Google.Apis.Services;
+using Xunit;
+using System.Threading.Tasks;
+using Google.Apis.Translate.v2.Data;
+using System.Linq;
+
+namespace IntegrationTests
+{
+    public class HttpClientFromMessageHandlerFactoryTests
+    {
+        public static IHostBuilder ConfigureHostBuilder(bool transientServiceClient) =>
+            Host.CreateDefaultBuilder()
+                .ConfigureServices(services =>
+                {
+                    // Configure one HttpClient that performs automatic decompression.
+                    services.AddHttpClient("allowsDecompression").ConfigurePrimaryHttpMessageHandler(() => new HttpClientHandler
```

Comment:
Ah... this is what I'd been missing. I'd thought we'd be mutating existing handlers, which made me really nervous.

Reply:
Nope, we just trust the client to the right thing and return what we are asking for. If they don't, we simply can't guarantee that service client calls will succeed later on.